agents/node/vcxagent-core/src/agent/service-connections.js

```diff
@@ -0,0 +1,184 @@
+    logger.info(`InviterConnectionSM has established connection ${connectionId}`)
+    return { invite, connection }
+  }
+
+  async function inviterConnectionCreateAndAccept (conenctionId, cbInvitation) {
+    const { invite, connection } = await inviterConnectionCreate(conenctionId, cbInvitation)
+    await _progressConnectionToAcceptedState(connection, 20, 2000)
+
+    const connSerialized = await connection.serialize()
+    await storeConnection(conenctionId, connSerialized)
+    logger.debug(`InviterConnectionSM after connection was accepted:\n${JSON.stringify(connSerialized)}`)
+    return { invite, connection }
+  }
+
+  // Invitee creates new connection from invite, sends connection request
+  async function inviteeConnectionAcceptFromInvitation (connectionId, invite) {
+    logger.info(`InviteeConnectionSM creating connection ${connectionId} from connection invitation.`)
+    const connection = await Connection.createWithInvite({ id: connectionId, invite })
+    logger.debug(`InviteeConnectionSM after created from invitation:\n${JSON.stringify(await connection.serialize())}`)
+    await connection.connect({ data: '{}' })
+    logger.debug('InviteeConnectionSM created connection agent')
+    await connection.updateState()
+
+    const connSerialized = await connection.serialize()
+    await storeConnection(connectionId, connSerialized)
+    logger.debug(`InviteeConnectionSM after sending connection request:\n${JSON.stringify(connSerialized)}`)
+    return connection
+  }
+
+  async function inviteeConnectionAcceptFromInvitationAndProgress (connectionId, invite) {
+    const connection = await inviteeConnectionAcceptFromInvitation(connectionId, invite)
+    await _progressConnectionToAcceptedState(connection, 20, 2000)
+    logger.info(`InviteeConnectionSM has established connection ${connectionId}`)
+    const connSerialized = await connection.serialize()
+    await storeConnection(connectionId, connSerialized)
+    return connection
+  }
+
+  async function _progressConnectionToAcceptedState (connection, attemptsThreshold, timeoutMs) {
+    async function progressToAcceptedState () {
+      if (await connection.updateState() !== StateType.Accepted) {
+        return { result: undefined, isFinished: false }
+      } else {
+        return { result: null, isFinished: true }
+      }
+    }
+
+    const [error] = await pollFunction(progressToAcceptedState, 'Progress connection', logger, attemptsThreshold, timeoutMs)
+    if (error) {
+      throw Error(`Couldn't progress connection to Accepted state. ${error}`)
+    }
+  }
+
+  async function connectionUpdate (connectionId) {
+    const connSerializedBefore = await loadConnection(connectionId)
+    const connection = await Connection.deserialize(connSerializedBefore)
+
+    const state = await connection.updateState()
+
+    const connSerialized = await connection.serialize()
+    await storeConnection(connectionId, connSerialized)
+
+    return state
```

Comment:
I would try to minimize all the blank lines and variable declarations which are used only once, as e.g. here, but that's just a nitpick. Also, maybe `loadConnection` could already perform the deserialization?